src/schema.rs

```diff
@@ -586,6 +592,30 @@ mod schema_tests {
                         "#),
                 "element_type"
         ),
+        case::annotations_constraint(
+                load(r#"
+                          b::d::5
+                          a::b::d::5
+                          b::c::d::5
+                          a::b::c::d::5
+                          b::a::d::5    // 'a' is treated as open content
+                          c::b::d::5       // 'c' is treated as open content
+                          c::b::a::d::5    // 'a' and 'c' are treated as open content
+                          open_content::open_content::b::d::5
+                          b::d::3.5
+                          b::d::"hello"
+                        "#),
+                load(r#"
+                          b::5
+                          d::5
+                          d::b::5
+                          5
+                        "#),
+                load_schema_from_text(r#" // For a schema with annotations constraint as below:
+                                type::{ name: annotations_type, annotations: ordered::optional::[a, required::b, c, required::d] }
```

Comment:
This is used within an `ion-schema-tests` and by default it will be considered as `optional` and treat this list-level optional as open content. But surely we don't want any improper ISL with open content here. So I can change this.

Reply:
I've created https://github.com/amzn/ion-schema-tests/pull/17 to fix it in `ion-schema-tests`.